app/models/form/sales/pages/joint_purchase.rb

```diff
@@ -2,11 +2,21 @@ class Form::Sales::Pages::JointPurchase < ::Form::Page
   def initialize(id, hsh, subsection)
     super
     @id = "joint_purchase"
-    @depends_on = [
-      { "ownershipsch" => 1 },
-      { "ownershipsch" => 2 },
-      { "companybuy" => 2 },
-    ]
+  end
+
+  def depends_on
+    if form.start_year_2025_or_later?
+      [
+         { "ownershipsch" => 1 },
+         { "ownershipsch" => 2 },
+       ]
+    else
+      [
+        { "ownershipsch" => 1 },
+        { "ownershipsch" => 2 },
+        { "companybuy" => 2 },
+      ]
+    end
   end
```

Comment:
I think it's these two pages that depend on the value from the "Is the buyer a company?" question. Changed the depends on.